libs/akita/src/lib/entityStore.ts

```diff
@@ -204,9 +218,17 @@ export class EntityStore<S extends EntityState = any, EntityType = getEntityType
         state,
         newStateOrFn,
         producerFn: this._producerFn,
+        onEntityIdChanges: (oldId: IDType, newId: IDType) => {
+          entityIdChanged = { oldId, newId };
+          this.entityIdChanges.next({ ...entityIdChanged, pending: true });
+        },
       })
     );
 
+    if (entityIdChanged) {
```

Comment:
Why not aggregate the ids and call it once here?

Reply:
Because undefined values would be emitted by the query when the id is updated: In fact the old entity is delete and thus undefined is emitted by the previous query before the new one emit the updated entity. This means the tracking query is notified about the id change (`pending=true`), so that undefined values can be suppressed and only then the updated entity is emitted again.